content/Software and Downloads/Upload/If-your-board-is-unresponsive-stuck-resetting-or-otherwise-not-working-as-expected.md

```diff
@@ -0,0 +1,32 @@
+---
+title: "If your board is unresponsive, stuck resetting, or otherwise not working as expected"
+---
+
+Learn what to do if your board appears to be running a sketch, but is not behaving as expected. If you're trying to upload a sketch but are getting an error in Arduino IDE, see [Errors when uploading a sketch](https://support.arduino.cc/hc/en-us/articles/4403365313810-Errors-when-uploading-a-sketch) instead.
+
+---
+
+1. Remove any connections to the RESET pin. If a black packaging foam is attached to the board, remove it.
+
+2. Check the power supply. If your board is powered by a USB cable, try using a different cable, or a different USB port on your computer. If using an external power supply, try using a USB cable.
+
+3. Check your sketch for incorrect use of [Watchdog](https://www.arduino.cc/reference/en/libraries/watchdog/) resets, or other declared reset functions, that can cause the board to reset unexpectedly. If you're able to upload your but it's not working as expected, debug your if, for and while statements with [Serial.print()](https://www.arduino.cc/reference/en/language/functions/communication/serial/print/) and check the Serial Monitor. Make sure you're not assigning values to an array outside its declared length.
```

Comment:
```suggestion
3. Check your sketch for incorrect use of [Watchdog](https://www.arduino.cc/reference/en/libraries/watchdog/) resets or other declared reset functions that can cause the board to reset unexpectedly. If you're able to upload your sketch but it's not working as expected, debug your if, for and while statements with [Serial.print()](https://www.arduino.cc/reference/en/language/functions/communication/serial/print/) and check the Serial Monitor. Make sure you're not assigning values to an array outside its declared length.
```

Removed the commas in the first sentence. Also, I think the word "sketch" was missing in the second sentence.